packages/template-retail-react-app/app/hooks/use-update-shopper-context.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
+import {
+    useUsid,
+    useShopperContext,
+    useShopperContextsMutation
+} from '@salesforce/commerce-sdk-react'
+import {isServer, isHydrated} from '@salesforce/retail-react-app/app/utils/utils'
+import {useQueryClient} from '@tanstack/react-query'
+
+// Hooks
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
+
+// Constants
+import {useShopperContextSearchParams} from '@salesforce/retail-react-app/app/hooks/use-shopper-context-search-params'
+
+/*
+ * This hook will set the shopper context when search params pertinant
+ * to shopper context are present.
+ */
+export const useUpdateShopperContext = () => {
+    const {site} = useMultiSite()
+    const {usid} = useUsid()
+    const {search} = useLocation()
+    const queryClient = useQueryClient()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {parameters: {usid, siteId: site.id}},
+        {enabled: !isServer}
+    )
+    console.log('shopperContext', shopperContext)
+
+    const refetchDataOnClient = () => {
+        queryClient.invalidateQueries()
+    }
+
+    const handleShopperContextUpdate = async (updateShopperContextObj) => {
+        if (Object.keys(updateShopperContextObj).length === 0) {
+            return
+        }
+
+        const payload = {
+            parameters: {usid, siteId: site.id},
+            body: updateShopperContextObj
```

Comment:
Should we validate the effectiveTimeDate value here? Will the API rejects if there is an invalidate effectiveDateTime?

Reply:
The API will reject with a 400 error for an invalid effectiveDateTime <img width="1035" alt="Screenshot 2024-09-12 at 11 06 15 AM" src="https://github.com/user-attachments/assets/0901dddb-3540-4091-a451-d22c17c3ed50">

Reply:
It is a good to validate the date before hitting the api. Here since it is in the search param, maybe show an error and redirect shopper to home page. (please talk to Trevor on the UX, thanks)

Reply:
I discussed this case with Trevor previously and he said that he hasn't seen cases where a message saying something didn't work pops up when clicking a link from an instagram/facebook promo. He and Nate confirmed that we should just show the site if it does fail.